### PR TITLE
Bugfix for automatic character width detection. Less garbage.

### PR DESCRIPTION
**Pull request by dluechoy**
Reads _Alpha_ instead of _Green_ channel when determining opacity. This allows automatic width detection to work with most fonts.
Automatic detection has been restricted to one character fullwidth, properties files will still override this.
Very large font textures (1024^2) will still have a single pixel-wide column of garbage at the edge of auto-detected fullwidth characters. (This now typically appears as dots after an 'M' or a '_' instead of "garbled" noise after anything wider than a '1')

The threshold for opacity has been moved to 16 from 0.

getStringWidthf resets bold ('link' ?) tracking for colour formatting codes as well, like the vanilla Minecraft fontRenderer outputs.

Signed-off-by: dluechoy dluechoy@gmail.com

### CODE REVIEW

File: newcode/src/com/pclewis/mcpatcher/mod/FontUtils.java [hunk 1]
```diff
@@ -73,6 +73,9 @@
             if (charWidthf[ch] <= 0.0f) {
                 charWidthf[ch] = 2.0f;
             }
+            else if (charWidthf[ch] <= 7.99f) {
```
Comment (dluechoy) [inline newcode/src/com/pclewis/mcpatcher/mod/FontUtils.java:78]: Damn typo, should be >=7.99 .

**Comment by prupe** [comment]
Thanks.  I merged these changes, and fixed another bug I found while looking at the the charWidth code.

**Comment by dluechoy** [comment]
There are a whole heap of bugs...
porting the fix for them to MCPatcher is on my to do list.
http://www.minecraftforum.net/topic/981282-white-text-fix/

**Comment by dluechoy** [comment]
https://github.com/pclewis/mcpatcher/issues/49/close

Accidentally pushed to same branch again.

**Pull request closed without merging**